brainglobe_workflows/brainmapper/analyse/analyse.py

```diff
@@ -265,19 +259,33 @@ def transform_points_downsampled_to_atlas_space(
     """
     field_scales = [int(1000 / resolution) for resolution in atlas.resolution]
     points: List[List] = [[], [], []]
+    points_out_of_bounds = []
     for axis, deformation_field_path in enumerate(deformation_field_paths):
         deformation_field = tifffile.imread(deformation_field_path)
         for point in downsampled_points:
-            point = [int(round(p)) for p in point]
-            points[axis].append(
-                int(
-                    round(
-                        field_scales[axis]
-                        * deformation_field[point[0], point[1], point[2]]
+            try:
+                point = [int(round(p)) for p in point]
+                points[axis].append(
+                    int(
+                        round(
+                            field_scales[axis]
+                            * deformation_field[point[0], point[1], point[2]]
+                        )
                     )
                 )
-            )
+            except IndexError:
+                if point not in points_out_of_bounds:
```

Comment:
To reduce memory usage, the code above transforms each point in x,y,z separately (otherwise 3 deformation fields need to be loaded at the same time). Without this if, each point would be potentially added to the list three times. 

Also I guess there could theoretically be duplicates if someone has edited the results manually.

Reply:
Ah I see, thanks!